server/services/store/sqlstore/migrations/000018_add_teams_and_boards.up.sql

```diff
@@ -274,137 +274,15 @@ CREATE TABLE {{.prefix}}boards_history (
-{{end}}
-{{if .mysql}}
-    UPDATE {{.prefix}}blocks as B
-INNER JOIN {{.prefix}}blocks as P
-        ON B.parent_id=P.id
-       SET B.board_id=P.id
-     WHERE P.type = 'board';
-
-    UPDATE {{.prefix}}blocks as B
-INNER JOIN {{.prefix}}blocks as P
-        ON B.parent_id=P.id
-INNER JOIN {{.prefix}}blocks as GP
-        ON P.parent_id=GP.id
-       SET B.board_id=GP.id
-     WHERE GP.type = 'board';
-
-    UPDATE {{.prefix}}blocks as B
-INNER JOIN {{.prefix}}blocks as P
-        ON B.parent_id=P.id
-INNER JOIN {{.prefix}}blocks as GP
-        ON P.parent_id=GP.id
-INNER JOIN {{.prefix}}blocks as GPP
-        ON GP.parent_id=GPP.id
-       SET B.board_id=GPP.id
-     WHERE GPP.type = 'board';
-
-    UPDATE {{.prefix}}blocks_history as B
-INNER JOIN {{.prefix}}blocks_history as P
-        ON B.parent_id=P.id
-       SET B.board_id=P.id
-     WHERE P.type = 'board';
-
-    UPDATE {{.prefix}}blocks_history as B
-INNER JOIN {{.prefix}}blocks_history as P
-        ON B.parent_id=P.id
-INNER JOIN {{.prefix}}blocks_history as GP
-        ON P.parent_id=GP.id
-       SET B.board_id=GP.id
-     WHERE GP.type = 'board';
-
-    UPDATE {{.prefix}}blocks_history as B
-INNER JOIN {{.prefix}}blocks_history as P
-        ON B.parent_id=P.id
-INNER JOIN {{.prefix}}blocks_history as GP
-        ON P.parent_id=GP.id
-INNER JOIN {{.prefix}}blocks_history as GPP
-        ON GP.parent_id=GPP.id
-       SET B.board_id=GPP.id
-     WHERE GPP.type = 'board';
-{{end}}
-{{if .postgres}}
-  UPDATE {{.prefix}}blocks as B
-     SET board_id=P.id
-    FROM {{.prefix}}blocks as P
-   WHERE B.parent_id=P.id
-     AND P.type = 'board';
-
-  UPDATE {{.prefix}}blocks as B
-     SET board_id=GP.id
-    FROM {{.prefix}}blocks as P,
-         {{.prefix}}blocks as GP
-   WHERE B.parent_id=P.id
-     AND P.parent_id=GP.id
-     AND GP.type = 'board';
-
-  UPDATE {{.prefix}}blocks as B
-     SET board_id=GGP.id
-    FROM {{.prefix}}blocks as P,
-         {{.prefix}}blocks as GP,
-         {{.prefix}}blocks as GGP
-   WHERE B.parent_id=P.id
-     AND P.parent_id=GP.id
-     AND GP.parent_id=GGP.id
-     AND GGP.type = 'board';
-
-  UPDATE {{.prefix}}blocks_history as B
-     SET board_id=P.id
-    FROM {{.prefix}}blocks_history as P
-   WHERE B.parent_id=P.id
-     AND P.type = 'board';
-
-  UPDATE {{.prefix}}blocks_history as B
-     SET board_id=GP.id
-    FROM {{.prefix}}blocks_history as P,
-         {{.prefix}}blocks_history as GP
-   WHERE B.parent_id=P.id
-     AND P.parent_id=GP.id
-     AND GP.type = 'board';
-
-  UPDATE {{.prefix}}blocks_history as B
-     SET board_id=GGP.id
-    FROM {{.prefix}}blocks_history as P,
-         {{.prefix}}blocks_history as GP,
-         {{.prefix}}blocks_history as GGP
-   WHERE B.parent_id=P.id
-     AND P.parent_id=GP.id
-     AND GP.parent_id=GGP.id
-     AND GGP.type = 'board';
-{{end}}
-
+UPDATE {{.prefix}}blocks as B SET B.board_id=B.root_id;
+UPDATE {{.prefix}}blocks_history as B SET B.board_id=B.root_id;
```

Comment:
This deletion look a bit dangerous. Do we not need all the old stuff because `B.root_id` points to the board already?

Reply:
All the deleted lines were to populate the board_id based on either the parent_id of a card, or the parent's parent for content blocks, down to parent's parent's parent.  However, cards and all children, plus views, have a root_id that points to the board.   There are no root_id is null anywhere in community which means we've been diligent in populating it.

Reply:
That's awesome! Always love when we can remove code 👍